drivers/pinctrl/pinctrl_stm32.c

```diff
@@ -122,21 +120,37 @@ static int stm32_pins_remap(const pinctrl_soc_pin_t *pins, uint8_t pin_cnt)
 
 	/* A valid remapping configuration is available */
 	/* Apply remapping before proceeding with pin configuration */
-	LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_AFIO);
+	if (!LL_APB2_GRP1_IsEnabledClock(LL_APB2_GRP1_PERIPH_AFIO)) {
+		LL_APB2_GRP1_EnableClock(LL_APB2_GRP1_PERIPH_AFIO);
+	}
 
 	if (STM32_REMAP_REG_GET(remap) == 0U) {
-		reg = &AFIO->MAPR;
+		/* read initial value, ignore write-only SWJ_CFG */
+		reg_val = AFIO->MAPR & ~AFIO_MAPR_SWJ_CFG;
+
+		/* set SWJ_CFG bits */
+#if defined(CONFIG_GPIO_STM32_SWJ_NONJTRST)
```

Comment:
Whole `choice GPIO_STM32_SWJ` should be removed from `drivers/gpio/Kconfig.stm32` and moved to `drivers/pinctrl/Kconfig.stm32` and `drivers/pinmux/Kconfig.stm32`.
This being said, this seems a bit over the kill as pinmux stuff is only there during a deprecation period and will be removed in next DV. Besides, we should consider a clear deprecation message for existing users, which is not easy in that situation.
I'm logging an issue to proceed with that change once pinmux has been removed.

Reply:
See https://github.com/zephyrproject-rtos/zephyr/issues/43640

Reply:
Thinking to it a bit more, I think we should convert this do device tree, similarly to what was done here: https://github.com/zephyrproject-rtos/zephyr/pull/39106

Reply:
Agree, the SWJ config within GPIO applies only to the F1. Therefore, it makes no sense to have it there.
Moved into F1 pinctrl dts.